common/stringop.c

```diff
@@ -309,3 +309,10 @@ char *argsep(char **stringp, const char *delim, char *matched) {
 	}
 	return start;
 }
+
+char *strict_regex(const char *str) {
+	size_t size = strlen(str) + 3;
+	char *reg = malloc(size);
```

Comment:
Need to check if allocation was successful. Otherwise, log an error and return `NULL`.